# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/jit-test/tests/cacheir/bug1842617.js
# Commit: 5af60a416884
# Full Hash: 5af60a4168846b5be7576faecbb5050fa6d965cf
# Author: Iain Ireland <iireland@mozilla.com>
# Date: 2023-07-13 09:17:48
# Description:
#   Bug 1842617: Check for empty shape list in branchTestObjShapeList r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D183192
# ==============================================================================

diff -r 6bab1aefd0bd -r 5af60a416884 js/src/jit-test/tests/cacheir/bug1842617.js
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/js/src/jit-test/tests/cacheir/bug1842617.js	Wed Jul 12 16:39:14 2023 +0000
@@ -0,0 +1,30 @@
+// |jit-test| --baseline-eager
+
+function foo(src) {
+  try {
+    evaluate(src);
+  } catch {}
+  Math = undefined;
+}
+
+foo(``);
+foo(``);
+foo(`var a = 0;`);
+foo(`var b = 0;`);
+foo(`var c = 0;`);
+foo(`var d = 0;`);
+foo(`undef();`);
+foo(`{`);
+foo(`
+   gc();
+   e = 0;
+   gc();
+`);
+foo(`
+   var f = 0;
+   gc();
+   delete Math;
+`);
+foo(`
+   g = undefined;
+`);